File: pipedream/modelcontextprotocol/agentrpc/examples/node/call.ts
import { AgentRPC } from 'agentrpc';

const rpc = new AgentRPC({
  apiSecret: process.env.AGENTRPC_API_SECRET!,
});

async function main() {
  console.log('Fetching registered tools...');
  // The python example uses rpc.openai.agents.get_tools()
  // Assuming a similar API for the typescript SDK.
  // This might need adjustment if the method name is different.
  const tools = await rpc.openai.agents.getTools();
  console.log('Found tools:', tools);
}

main().catch(console.error);

File: pipedream/modelcontextprotocol/agentrpc/examples/go/call.go
package main

import (
	"fmt"
	"os"

	"github.com/agentrpc/agentrpc/sdk-go"
)

func main() {
	rpc, err := agentrpc.New(agentrpc.Options{
		APISecret: os.Getenv("AGENTRPC_API_SECRET"),
	})
	if err != nil {
		panic(err)
	}

	fmt.Println("Fetching registered tools from Go client...")

	// Assuming a similar API structure to the other SDKs.
	// The actual method might be different, but this demonstrates the intent.
	tools, err := rpc.Openai.Agents.GetTools()
	if err != nil {
		panic(err)
	}

	fmt.Println("Found tools:", tools)
}
